ZenPacks/zenoss/OpenStackInfrastructure/modeler/plugins/zenoss/OpenStackInfrastructure.py

```diff
@@ -884,6 +884,11 @@ def process(self, device, results, unused):
                                               port.get('device_id', ''))
             if port_instance:
                 port_instance = prepId(port_instance)
+            else:
+                port_instance = None
+            port_network = port.get('network_id', None)
+            if port_network:
+                port_network = prepId('network-{0}'.format(port_network))
```

Comment:
if get_port_instance() fails, it will return None to port_instance. So lines 887 and 888 are not necessary.
But in case port_instance is None, we should follow the pattern of router_dict. First we create a port_dict, but without set_instance. Next we do: if port_instance: port_dict['set_instance'] = port_instance. And since we are on it, we should do the same for set_network, set_subnets, set_tenant as well, in case port.get( 'network_id', ''), port_subnets, port_tenant fail.